Signal/src/ViewControllers/MessageDetailViewController.swift

```diff
@@ -343,25 +352,22 @@ class MessageDetailViewController: OWSViewController, UIScrollViewDelegate {
             self.messageTextTopConstraint = messageTextView.autoPinEdge(toSuperviewEdge: .top, withInset: 0)
             self.messageTextHeightLayoutConstraint = messageTextView.autoSetDimension(.height, toSize:0)
 
-            let leadingMargin: CGFloat = isIncoming ? 15 : 10
-            let trailingMargin: CGFloat = isIncoming ? 10 : 15
-
             let bubbleView = UIImageView(image: bubbleImageData.messageBubbleImage)
             self.bubbleView = bubbleView
 
             bubbleView.layer.cornerRadius = 10
             bubbleView.addSubview(messageTextProxyView)
 
-            messageTextProxyView.autoPinEdge(toSuperviewEdge: .leading, withInset: leadingMargin)
-            messageTextProxyView.autoPinEdge(toSuperviewEdge: .trailing, withInset: trailingMargin)
+            messageTextProxyView.autoPinEdge(toSuperviewEdge: isIncoming ? .leading : .trailing, withInset: messageTailEdgeMargin)
+            messageTextProxyView.autoPinEdge(toSuperviewEdge: isIncoming ? .trailing : .leading, withInset: messageNoTailEdgeMargin)
             messageTextProxyView.autoPinHeightToSuperview(withMargin: 10)
             self.messageTextProxyViewHeightConstraint = messageTextProxyView.autoSetDimension(.height, toSize:0)
 
             let row = UIView()
             row.addSubview(bubbleView)
             bubbleView.autoPinHeightToSuperview()
-            bubbleView.autoPinLeadingToSuperview(withMargin: 10)
-            bubbleView.autoPinTrailingToSuperview(withMargin: 10)
+            bubbleView.autoPinEdge(toSuperviewEdge: isIncoming ? .leading : .trailing, withInset: bubbleViewHMargin)
+            self.bubbleViewWidthConstraint = bubbleView.autoSetDimension(.width, toSize:0)
```

Comment:
Why are we setting the width to 0?

Reply:
This is a placeholder value. We set the actual value in `updateTextLayout()` which does measurement, etc..